### DISCUSSION TITLE
What are the max sizes I can use imagemagick for?

### DISCUSSION THREAD

**Discussion by merspieler** [ImageMagick/ImageMagick#2626] [Help] [answered]
I want to batch process multiple 4k (4096x4096) files into a single 16k (16384x16384).

Right now, this fails, probably due to size.

What are the max sizes, I can work with?

My version is ImageMagick 6.9.11-24 Q16 x86_64 20200718

Does a newer version change that limit?

**Comment by snibgo** [comment, marked as answer]
What is your command? How does it fail? What error messages?

IM does not have itself a low limit on image dimensions. "convert -list resource" may show a limit on your installation. If so, then edit policy.xml to suit.

For example:
```
convert -list resource

Resource limits:
  Width: 214.7MP
  Height: 214.7MP
  List length: 18.446744EP
  Area: 6.4068GP
  Memory: 2.9834GiB
  Map: 5.9668GiB
  Disk: unlimited
  File: 1536
  Thread: 8
  Throttle: 0
  Time: unlimited
```

For good performance, your memory should be sufficient that the pixels do not need to be cached on disk. Assuming your IM is not HDRI, Q16 v6 needs 8 bytes per pixel, so your 16k x 16k images need 2.15 GB of memory.

EDIT to add: perhaps you are reading all your 4k images into memory, and enlarging them all in memory, in a single command. If so, then that's the problem. Don't do them all together.

**Reply by merspieler** [reply]
Okay, so it seems like I'm here at/over the limit:
```
$ convert -list resource
Resource limits:
  Width: 16KP
  Height: 16KP
  List length: unlimited
  Area: 128MP
  Memory: 256MiB
  Map: 512MiB
  Disk: 1GiB
  File: 768
  Thread: 4
  Throttle: 0
  Time: unlimited
```

Where do I find that policy.xml?

Commands are
* `montage <4k files>.png -geometry +0+0 -tile 4x4 <16k file>.png`
* `convert <16k>.png -flip <16k>.png`

montage failed with 
```
montage-im6.q16: width or height exceeds limit `r1.png' @ error/cache.c/OpenPixelCache/3897
```

On my dev machine I've got 16GB RAM, on the server I'll have even more.

**Reply by fmw42** [reply]
See https://imagemagick.org/script/resources.php. It lists the possible places. But /etc is usually the one you want.

**Reply by merspieler** [reply]
awesome, workt.

Thanks guys.

**Reply by SirMotte** [reply]
Sorry for exhuming this thread.

Issue:
- Large image files (8818x19836) can't be converted from .jpg or .png to .webp
- changes to the policy.xml do not register
- the resource prompt shows width and height as pp, not kp

Information:
- Windows 11 21H2 (Build 22000.1042)
- 16GB Ram
- Lots and Lots of Disk Space
- ImageMagick-7.1.0-Q16-HDRI

The error message is as follows:
magick: width or height exceeds limit `largetestimage8818x19836.webp' @ error/webp.c/WriteWEBPImage/1146.

![grafik](https://user-images.githubusercontent.com/82598692/195904559-a33b797c-6a7c-4966-b940-1059b56d5e1b.png)

Any help on what I'm doing wrong would be highly appreciated.

**Comment by snibgo** [comment]
IM can't write WEBP images that are wider or taller than 16383 pixels. This limit seems to be imposed by WEBP.

**Reply by SirMotte** [reply]
> IM can't write WEBP images that are wider or taller than 16383 pixels. This limit seems to be imposed by WEBP.

This is enlightening, thank you very much for the quick response!

Edit:
Jupp, confirmed in the google dev FAQ
![grafik](https://user-images.githubusercontent.com/82598692/195912786-ceeff78c-c231-4a05-a6ff-70bdad63c704.png)
Thanks again, I was on the verge of giving up.

**Comment by snibgo** [comment]
> the resource prompt shows width and height as pp, not kp

"PP" mean "peta pixels". One peta is 10^15. So your width and height limits are 461 thousand million million pixels. However, your area (width \* height) is limited to 17\*10^9 pixels, so that is a more restrictive limit.

**Reply by SirMotte** [reply]
Thank you for clarifying, I'm slowly getting the hang of it.
